DigiMeSDK/Core/Classes/Network/DMERequestFactory.m

```diff
@@ -80,6 +83,88 @@ - (NSURLRequest *)fileRequestWithId:(NSString *)fileId sessionKey:(NSString *)se
+    
+    [request setCachePolicy:NSURLRequestReloadIgnoringLocalCacheData];
+    [request setHTTPMethod:@"POST"];
+    
+    NSString *boundary = [self generateBoundaryString];
+    NSString *metadata = [[headers[@"metadata"] stringByReplacingOccurrencesOfString:@"\n" withString:@""]stringByReplacingOccurrencesOfString:@"\r" withString:@""];
+    NSString *symmetricalKey = [[headers[@"symmetricalKey"] stringByReplacingOccurrencesOfString:@"\n" withString:@""]stringByReplacingOccurrencesOfString:@"\r" withString:@""];
+    
+    NSString *contentType = [NSString stringWithFormat:@"multipart/form-data; boundary=%@", boundary];
+    [request setValue:contentType forHTTPHeaderField: @"Content-Type"];
+    [request setValue:@"application/json" forHTTPHeaderField:@"Accept"];
+    [request setValue:headers[@"sessionKey"] forHTTPHeaderField: @"sessionKey"];
+    [request setValue:symmetricalKey forHTTPHeaderField: @"symmetricalKey"];
+    [request setValue:headers[@"iv"] forHTTPHeaderField: @"iv"];
+    [request setValue:metadata forHTTPHeaderField: @"metadata"];
+    
+    NSMutableData *body = [NSMutableData data];
+    [body appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
+    [body appendData:[@"Content-Type: multipart/form-data\r\n" dataUsingEncoding:NSUTF8StringEncoding]];
+    [body appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=%@; filename=%@\r\n\r\n", @"file", @"file"] dataUsingEncoding:NSUTF8StringEncoding]];
+    
+    if (data)
+    {
+        [body appendData:data];
+        [body appendData:[[NSString stringWithFormat:@"\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
+    }
+    
+    [body appendData:[[NSString stringWithFormat:@"--%@--\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
+    [request setHTTPBody:body];
+    NSString *postLength = [NSString stringWithFormat:@"%lu", [body length]];
+    [request setValue:postLength forHTTPHeaderField:@"Content-Length"];
+    
+    return request;
+}
+
+- (NSData *)createBodyWithBoundary:(NSString *)boundary
+                        parameters:(NSDictionary *)parameters
+                              file:(NSData *)data
+{
+    NSMutableData *httpBody = [NSMutableData data];
+    
+    [parameters enumerateKeysAndObjectsUsingBlock:^(NSString *parameterKey, NSString *parameterValue, BOOL *stop) {
+        [httpBody appendData:[[NSString stringWithFormat:@"--%@\r\n", boundary] dataUsingEncoding:NSUTF8StringEncoding]];
+        [httpBody appendData:[[NSString stringWithFormat:@"Content-Disposition: form-data; name=\"%@\"\r\n\r\n", parameterKey] dataUsingEncoding:NSUTF8StringEncoding]];
+        [httpBody appendData:[[NSString stringWithFormat:@"%@\r\n", parameterValue] dataUsingEncoding:NSUTF8StringEncoding]];
+    }];
+    
+    NSString *mimetype = @"application/json";//[self mimeTypeForPath:path];
```

Comment:
Is the code meant to be commented? The mime type in that case appears to be hardcoded to `application/json`, and the commented method isn't used.